src/utils/env.ts

```diff
@@ -65,3 +87,24 @@ export const lookupPcEnv = (pcEnvTld: string, lenient = false, envTlds: string[]
 
     return null;
 };
+
+/**
+ * Attempts to locate a PC environment corresponding to the provided url
+ * @param url A string representing the Genesys Cloud environment top-level domain to search for
+ * @returns A Genesys Cloud environment object if found; null otherwise.
+ */
+export const lookupGcEnv = (url: string, targetEnv: string, parseEnvironment: EnvironmentParser = parse): PcEnv|null => {
+    try {
+        const hostLocation = new URL(url);
+        if (['localhost', '127.0.0.1'].includes(hostLocation.hostname)) {
+            return {
+                pcEnvTld: 'localhost',
+                pcAppOrigin: hostLocation.origin
+            };
+        } else {
+            return findPcEnvironment(hostLocation, targetEnv, parseEnvironment);
+        }
+    } catch {
```

Comment:
nice!  glad you wrapped this entire block because the `parse` method can throw if the url is invalid.

Which makes me realize we should add a test for that case so we're sure we return null (which eventually throws) if we're passed a malformed url for gcHostOrigin